### PR TITLE
Improve tiered compaction tests

### PR DESCRIPTION
**Pull request by arpad-m**
Improves the tiered compaction tests:

* Adds a new test that is a simpler version of the ignored `test_many_updates_for_single_key` test.
* Reduces the amount of data that `test_many_updates_for_single_key` processes to make it execute more quickly. 
* Adds logging support.

### CODE REVIEW

File: pageserver/compaction/tests/tests.rs [hunk 1]
```diff
@@ -33,3 +49,26 @@ async fn test_many_updates_for_single_key() {
...
+        println!("layer {}: {}", l.short_id(), l.file_size());
+    }
+
+    println!("Running compaction...");
+    executor.compact().await.unwrap();
+
+    for l in executor.live_layers.iter() {
+        println!("layer {}: {}", l.short_id(), l.file_size());
+    }
+}
```
Comment (koivunej) [inline pageserver/compaction/tests/tests.rs:74]: Please add a newline here.

File: pageserver/compaction/tests/tests.rs [hunk 2]
```diff
@@ -11,13 +26,14 @@ use pageserver_compaction::simulator::MockTimeline;
 #[ignore]
 #[tokio::test]
 async fn test_many_updates_for_single_key() {
+    setup_logging();
```
Comment (koivunej) [inline pageserver/compaction/tests/tests.rs:29]: Is the test comment still up to date?

**Review by koivunej — APPROVED** [review]
Approved without the whitespace error.

File: pageserver/compaction/tests/tests.rs (hunk 1 above)
Comment (koivunej) [reply pageserver/compaction/tests/tests.rs:74]: Rustfmt agrees :)

Comment (arpad-m) [reply pageserver/compaction/tests/tests.rs]: Added it.

**Pull request merged**